src/modules/coinmarket/coinMarketClient.ts

```diff
@@ -0,0 +1,87 @@
+/*******************************************************************************
+
+    This file contain CoinMarket Interface and CoinGeckoMaket as concrente implementation of CoinMarket.
+
+    Copyright:
+        Copyright (c) 2020-2021 BOSAGORA Foundation
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+import { CoinGeckoClient } from "coingecko-api-v3";
+import { logger, Logger } from '../common/Logger';
+import { IMarketCap } from "../../Types";
+
+export interface CoinMarket {
+    /**
+     * Method to ping the coin market
+     */
+    ping(): Promise<boolean>;
+    /**
+     * Method to fetch from coin market
+     */
+    fetch(): Promise<any>;
+}
+
+export class CoinGeckoMaket implements CoinMarket {
+    /**
+     * Market Client i.e Geco_coin_market 
+     */
+    private coinMarketClient: CoinGeckoClient;
+
+    constructor() {
+        this.coinMarketClient = new CoinGeckoClient({
+            timeout: 10000,
+            autoRetry: true,
+        });
+    }
+    /**
+     *  This method ping the GecoCoinMarket
+     * @returns Returns the Promise. If it is finished successfully the `.then`
+     * of the returned Promise is called and if an error occurs the `.catch`
+     * is called with an error.
+     */
+    public ping(): Promise<boolean> {
+        return new Promise<boolean>(async (resolve, reject) => {
+            let data = await this.coinMarketClient.ping();
+            if (data.gecko_says) {
+                resolve(true);
+            }
+            else {
+                logger.info(`Error: Unable to ping coin market client`);
+                return reject(`Error: Unable to ping coin market client`);
+            }
+        })
+    }
+    /**
+     *  This method fetch the coin market data
+     * @returns Returns the Promise. If it is finished successfully the `.then`
+     * of the returned Promise is called and if an error occurs the `.catch`
+     * is called with an error.
+     */
+    public fetch(): Promise<any> {
+        return new Promise<any>(async (resolve, reject) => {
+            let marketCap = await this.coinMarketClient.simplePrice({ ids: "boa", vs_currencies: "usd", include_market_cap: true, include_24hr_vol: true, include_24hr_change: true, include_last_updated_at: true })
+            if (marketCap) {
+                let coinMarketStat: IMarketCap =
+                {
+                    usd_price: marketCap.boa.usd,
+                    usd_market_cap: marketCap.boa.usd_market_cap,
+                    usd_24h_vol: marketCap.boa.usd_24h_vol,
+                    usd_24h_change: marketCap.boa.usd_24h_change,
+                    last_updated_at: marketCap.boa.last_updated_at
+                }
```

Comment:
We support 3 languages on the screen.
Calls should be supported according to language changes.
English:USD, Korean:KRW, Chinese:CNY
This will require currency type on the table.

Reply:
we will use socket to and catch on laguage change from frontend. i will create seprate PR for sockets